python/graphstorm/eval/eval_func.py

```diff
@@ -192,62 +194,112 @@ def eval_roc_auc(logits,labels):
     predicted_labels=logits
     predicted_labels=predicted_labels.detach().cpu().numpy()
     labels=labels.detach().cpu().numpy()
+
+    # check if the two inputs have the same number of rows.
+    try:
+        assert predicted_labels.shape[0] == labels.shape[0]
+    except AssertionError:
+        logging.error('Predictions and labes should have ' + \
+                f'the same number of records, but got, {predicted_labels.shape[0]} and ' + \
+                f'{labels.shape[0]}.')
+        raise
+
+    # check if the predictions >=2D.
+    try:
+        assert len(predicted_labels.shape) > 1
+    except AssertionError:
+        logging.error('GraphStorm assumes the predicted logit is a 2D tesnor, ' + \
+                      'but got a 1D tensor.')
+        raise
+
     # The roc_auc_score function computes the area under the receiver operating characteristic
     # (ROC) curve, which is also denoted by AUC or AUROC. The following returns the average AUC.
     rocauc_list = []
-    labels=labels_to_one_hot(labels, predicted_labels.shape[1])
-    for i in range(labels.shape[1]):
-        # AUC is only defined when there is at least one positive data.
-        if np.sum(labels[:, i] == 1) > 0 and np.sum(labels[:, i] == 0) > 0:
-            is_labeled = labels[:, i] == labels[:, i]
-            rocauc_list.append(roc_auc_score(labels[is_labeled, i],
-                                             predicted_labels[is_labeled, i]))
+    # For binary results, the sklearn roc_auc_score function asks 1D inputs of predictions.
+    # And the label is a 1D tensor. For other cases, the sklearn roc_auc_score function asks
+    # nD inputs of predictions. So here we need to check the predictions' 2nd dim for the binary
+    # conditions.
+    if len(predicted_labels.shape)>1 and predicted_labels.shape[1] == 2 \
```

Comment:
We have an assert in L209, we do not need the `len(predicted_labels.shape)>1` check here.